### PR TITLE
fix: stream ownership check

### PR DESCRIPTION
**Pull request by vlad-diachenko**
**What this PR does / why we need it**:

It has changed how the stream is checked if the ingester owns it. Now, the behavior is exactly the same as on the distributor.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: pkg/ingester/owned_streams.go [hunk 1]
```diff
@@ -11,11 +11,11 @@ import (
 	"github.com/grafana/loki/v3/pkg/util/constants"
 )
```
Comment (JordanRushing) [inline pkg/ingester/owned_streams.go:13]: Probably worth adding a metric here to track the duration for recalculation.

File: pkg/ingester/recalculate_owned_streams.go [hunk 2]
```diff
@@ -42,34 +39,50 @@ func (s *recalculateOwnedStreams) iteration(_ context.Context) error {
...
+	for _, instance := range s.instancesSupplier() {
+		oldLimit, newLimit := instance.ownedStreamsSvc.updateFixedLimit()
+		if oldLimit != newLimit {
+			level.Info(s.logger).Log("msg", "fixed limit has been updated", "tenant", instance.instanceID, "old", oldLimit, "new", newLimit)
 		}
-		instance.ownedStreamsSvc.updateFixedLimit()
 	}
 }
 
+func (s *recalculateOwnedStreams) resetRingState() {
```
Comment (JordanRushing) [inline pkg/ingester/recalculate_owned_streams.go:80]: We might not need this method any more?

File: pkg/ingester/owned_streams.go (hunk 1 above)
Comment (vlad-diachenko) [reply pkg/ingester/owned_streams.go:13]: done

**Review by JordanRushing — APPROVED** [review]
LGTM!

**Pull request merged**